src/emu/layout/s3dglass.lay

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0"?>
+<!--
+license:CC0
+-->
+<mamelayout version="2">
+	<view name="SegaScope Left LCD Standard (4:3)">
+		<screen tag="left_lcd">
+			<bounds left="0" top="0" right="4" bottom="3" />
+		</screen>
+	</view>
+
+	<view name="SegaScope Right LCD Standard (4:3)">
+		<screen tag="right_lcd">
+			<bounds left="0" top="0" right="4" bottom="3" />
+		</screen>
+	</view>
+
+	<view name="SegaScope Left LCD Aspect (~scr1nativexaspect~:~scr1nativeyaspect~)">
+		<screen tag="left_lcd">
+			<bounds left="0" top="0" right="~scr1width~" bottom="~scr1height~" />
+		</screen>
+	</view>
+
+	<view name="SegaScope Right LCD Pixel Aspect (~scr1nativexaspect~:~scr1nativeyaspect~)">
+		<screen tag="right_lcd">
+			<bounds left="0" top="0" right="~scr1width~" bottom="~scr1height~" />
+		</screen>
+	</view>
```

Comment:
There’s no point adding these views – MAME always generates single-screen views at physical and square pixel aspect.  Also, what you’re doing can break if the device is used as part of a larger system setup because the the screen numbers and tags might not line up the way you expect.

Reply:
I tested without these views and MAME just created two more single-screen views for each glasses, one for left LCD and another for right LCD, both at pixel aspect (67:56) only, but called them "Standard". Also, the LCDs were only identified numerically as screens, using Screen 1 and Screen 2 for one glasses, combined with Screen 3 and Screen 4 when two glasses are plugged-in (possible with the smsj driver). With the views above, using tags, the screens are identified correctly and the path to the related 3-D glasses is informed.

Unfortunately there is still a problem and all screens are still listed below the custom views, numerically (0 to 2, or 0 to 4) with screens above 0 using the (67:56) ratio described as Standard. Maybe because of this problem MAME initially creates a default view with all screens shown at once. To avoid this problem I had to kept the old layout file for sms1 (src/mame/layout/sms1.lay), that had a workaround to hide screens 1 and 2 by default, and added another view to also hide screens 1 up to 4 and two glasses are used. The code of an old branch I was using ignored the views with empty screens for glasses when they were not used, but now, with a more up to date branch, I noticed some errors are informed at command line when glasses are not used, but the execution continues normally:

Error instantiating layout view Main Screen Standard (4:3): invalid screen index 1
Error instantiating layout view Main Screen (2x3D) Standard (4:3): invalid screen index 1
Error instantiating layout view Main Screen Standard (4:3): invalid screen index 1
Error instantiating layout view Main Screen (2x3D) Standard (4:3): invalid screen index 1

This is not pretty. Could you take a look at the layout code to ignore screens added by non fixed devices? Or maybe add a parameter to set a view by default, that tells MAME to use it by default instead try to created a view with all screens at once.

Reply:
The implementation was (still is) wrong, isn't it? The glasses don't have 2 lcd screens in them, but electronic shutters. I've emulated shutters in tceptor2(tceptor.cpp). It's a flickerfest but I can see the 3D effect when I set the view to crossed view and then cross my eyes.

Reply:
@happppp I saw how the implementation of the shutters in tceptor2 was done by the time you posted about it in issue 3492. I didn't know you are the author. I glad about how you are proud of that implementation, but it is not useful to me. I want to be able to reproduce the effect the way the hardware would reproduce it, accurately, but I also want to allow the users to play the game without punish them. If you haven't noticed yet, in issue 3492 the user was only able to enjoy the 3D effect of the game with a modern hardware using my "binocular hack". If there was a better way to blit the previous frame seen through a lens to the same lens then I wouldn't need to simulate them with screens. Until then, I'm not changing my approach.

By the way, please think about removing that crossed view, it's not the way the 3D effect is obtained, and I hope you will not prejudice your eyes by crossing them for long periods.

Reply:
The side by side straight/crossed views are there for convenience. The best option would be to output to VR goggles (assuming MAME can output to them as 2 different screens). Left screen view to "3D Scope Left", right screen view to "3D Scope Right".

Reply:
The side by side straight view is not just convenience, it works as a basic representation of what is been seen through glasses, helps to understand how the effect works, but the crossed view is wrong. Looking down a corridor, the left eye should see more of the right wall and the right eye should see more of the left wall.

Reply:
I'm ok with our disagreement on that.
My main nitpick is the addition of 2 fake screen devices.

Reply:
I just used the tool that worked for what I needed. If there is a better tool, I will change the implementation, but keep in mind that I also want to play the games. The "binocular hack" allows greater flexibility, I can play using any side in full screen, using the side by side on my PC or I can try the side by side with a 3D TV or other device that can translate the view into a 3D effect. I think about adding notes to the new 3D glasses device to better document that the screens are fake and the real device use shutters.